admin_bot/view_schedule/handlers.py

```diff
@@ -78,8 +89,41 @@ def show_trainingroupday_info(update, context: CallbackContext):
     players_info = f"{group_name}{group_level}{group_players}{visitors}{pay_visitors}{pay_bonus_visitors}{absents}"
     text = f"{general_info}{players_info}"
 
-    markup = back_from_show_grouptrainingday_info_keyboard(
-        year=tr_day.date.year, month=tr_day.date.month, day=tr_day.date.day
+    markup = show_grouptrainingday_info_keyboard(tr_day)
+
+    bot_edit_message(context.bot, text, update, markup)
+
+
+def show_confirm_train_action(update: Update, context: CallbackContext):
+    query = update.callback_query
+    (purpose, action, year, month, day, start_time) = separate_callback_data(query.data)
+    markup = show_grouptrainingday_confirm_keyboard(
+        year, month, day, start_time, action
     )
 
+    if action != GroupTrainingDay.AVAILABLE:
+        text = "Подтверди отмену занятия"
+    else:
+        text = "Подтверди доступность занятия"
+
     bot_edit_message(context.bot, text, update, markup)
+
+
+def train_action_handler(update: Update, context: CallbackContext):
+    query = update.callback_query
+    (purpose, action, year, month, day, start_time) = separate_callback_data(query.data)
+
+    tr_day = GroupTrainingDay.objects.filter(
+        date__year=year, date__month=month, date__day=day, start_time=start_time
+    ).first()
+
+    if tr_day != None:
```

Comment:
```suggestion
    if tr_day:
```
Некорректно сравнивать с `None` через оператор `!=`. Нужно через `is not`